codegen-core/common-test-models/constraints.smithy

```diff
@@ -0,0 +1,441 @@
+$version: "1.0"
+
+namespace com.amazonaws.constraints
+
+use aws.protocols#restJson1
+
+/// A service to test aspects of code generation where shapes have constraint traits.
+@restJson1
+@title("ConstraintsService")
+service ConstraintsService {
+    operations: [
+      ConstrainedShapesOperation,
```

Comment:
Should these operation names adhere to the same standard as the Pokemon service example? 

> {Verb}[{Qualifier}]{Noun}

Reply:
I'm struggling coming up with reasonable suggestions for `Verb` and `Noun` for such academic examples.